frontend/src/components/rubric-builder/RatingInput.tsx

```diff
@@ -40,33 +51,74 @@ export default function RatingInput({
   };
 
   return (
-    <div className={'grid grid-rows-1 grid-col-3 grid-flow-col gap-2 w-full'}>
-      <input
-        type="number"
-        value={ratingValue} // use local state for value
-        onChange={handlePointChange} // properly handle points change
-        className="hover:bg-gray-800 rounded-lg p-3 text-gray-300 w-16 border border-gray-600 bg-gray-500 focus:outline-none focus:ring-2 focus:ring-blue-500"
-        min="0"
-        required
-      />
-      <input
-        type="text"
-        className="hover:bg-gray-800 rounded-lg p-3 text-gray-300 border border-gray-600 bg-gray-500 focus:outline-none focus:ring-2 focus:ring-blue-500"
-        placeholder="Enter rating description..."
-        value={ratingDescription} // use local state for value
-        onChange={handleDescriptionChange} // properly handle description change
-      />
-      <button
-        className={
-          'bg-gray-200 text-black px-2 py-1 rounded opacity-20 hover:bg-red-500 hover:opacity-100' +
-          ' hover:text-white'
-        }
-        tabIndex={-1}
-        onClick={handleRemoveRatingPress} // properly handle the remove button
-        type={'button'}
-      >
-        -
-      </button>
+    <div className="grid grid-rows-1 grid-cols-1 gap-2 w-full">
+      <div className="grid grid-cols-[0.5fr_2.5fr_0.2fr] gap-2 w-full">
+        <input
+          type="number"
+          value={ratingValue} // use local state for value
+          onChange={handlePointChange} // properly handle points change
+          className="hover:bg-gray-800 rounded-lg p-2 text-gray-300 w-12 border border-gray-600 bg-gray-500 focus:outline-none focus:ring-2 focus:ring-blue-500"
+          min="0"
+          required
+        />
+        <input
+          type="text"
+          className="hover:bg-gray-800 rounded-lg p-3 text-gray-300 border border-gray-600 bg-gray-500 focus:outline-none focus:ring-2 focus:ring-blue-500 flex-grow"
+          placeholder="Enter rating description..."
+          value={ratingDescription} // use local state for value
+          onChange={handleDescriptionChange} // properly handle description change
+        />
+        <button
+          className="bg-gray-200 text-black px-2 py-1 rounded opacity-20 hover:bg-red-500 hover:opacity-100 hover:text-white"
+          tabIndex={-1}
+          onClick={handleRemoveRatingPress} // properly handle the remove button
+          type="button"
+        >
+          -
+        </button>
+      </div>
+
+      {/* Long description field below the main description field */}
+      <div className="col-span-3">
+        <button
+          className="text-blue-500 hover:underline"
+          onClick={() => setIsPopupOpen(true)}
+          type="button"
+        >
+          Edit Long Description
+        </button>
+      </div>
+
+      {/* Popup for editing the long description */}
+      {isPopupOpen && (
+        <div className="fixed inset-0 bg-black bg-opacity-50 flex justify-center items-center">
```

Comment:
There looks to be an issue with the z-axis position of the popup. Only the popup from the most recently added rating group stays in front of everything.

![image](https://github.com/user-attachments/assets/d20760ef-9cd4-405c-a008-88ffa915196d)

![image](https://github.com/user-attachments/assets/cdf1e194-179c-4d58-bbbb-6905e3b2eb26)

Reply:
Still need to fix this bug and merge in latest dev